src/api/workspace.ts

```diff
@@ -130,3 +135,55 @@ export async function waitForBuild(
 	);
 	return updatedWorkspace;
 }
+
+/**
+ * Streams agent logs to the emitter in real-time.
+ * Fetches existing logs and subscribes to new logs via websocket.
+ * Returns the websocket and a completion promise that rejects on error.
+ */
+export async function streamAgentLogs(
+	client: CoderApi,
+	writeEmitter: vscode.EventEmitter<string>,
+	agent: WorkspaceAgent,
+): Promise<{
+	socket: OneWayWebSocket<WorkspaceAgentLog[]>;
+	completion: Promise<void>;
+}> {
+	// This fetches the initial bunch of logs.
+	const logs = await client.getWorkspaceAgentLogs(agent.id);
+	for (const log of logs) {
+		writeEmitter.fire(log.output + "\r\n");
+	}
```

Comment:
The watch endpoint already streams all the logs, I think (including previous ones, since the default for the `after` query param is zero which means all logs), so I believe we would not need the extra `getWorkspaceAgentLogs` call.